### DISCUSSION TITLE
Using mpv debug plugin in mpv.net

### DISCUSSION THREAD

**Discussion by tsl0922** [mpvnet-player/mpv.net#622] [Ideas]
[mpv-debug-plugin](https://github.com/tsl0922/mpv-debug-plugin) is a visual debug tool for mpv, you can use it to view mpv's internal properties / logs and run mpv command, very useful for script developers.

To use mpv-debug-plugin in mpv.net:

- download the plugin from https://github.com/tsl0922/mpv-debug-plugin/releases
- place `debug.dll` in `scripts` folder under `Settings - Open Config Folder`
- `Settings - Edit input.conf`, add `` ` script-message-to debug show  #custom-menu: Debug ``

After restarting mpv.net, you will be able to open the Debug window via the `` ` `` hotkey or the `Custom - Debug` menu.

![](https://github.com/tsl0922/mpv-debug-plugin/raw/main/screenshot/debug.jpg)

**Comment by dyphire** [comment]
Although I really like the idea of using plugins, using IMGUI as the UI is not a good idea. One is that the interface display of imgui is very disconnected from other interfaces of mpvnet, and the other is that the font loading and rendering of imgui are very poor, with poor support for CJK characters.

**Reply by tsl0922** [reply]
Well, I agree with you.

But it's not worth to develop it again with another UI framework, as it is just a developer tool.
